### PR TITLE
Login gplus

### PR DESCRIPTION
**Pull request by mvn-vyle-dn**
## Description
- login via gmail
- modify show welcome when login success
- change domain to: [foody.com](http://foody.com)
## Reference Requirements
- Task ID: TI-1

## Reference Isuees
#{issue_id}

## Migrations
YES | NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: database/migrations/2018_07_23_072844_create_users_table.php [hunk 1]
```diff
@@ -22,6 +22,7 @@ public function up()
             $table->string('phone');
             $table->string('email')->unique();
             $table->string('password');
+            $table->string('gplus_id')->nullable();
```
Comment (at-quangtran) [inline database/migrations/2018_07_23_072844_create_users_table.php]: cái này là viết 1 migration mới add thêm cột vô chứ ko sửa ở file cũ

File: public/js/public/logout.js [hunk 2]
```diff
@@ -12,9 +12,14 @@ $(document).ready(function () {
...
           localStorage.removeItem('token-login');
+          signOut();
           window.location.href = 'http://' + window.location.hostname;
         }
       });
     }
   });
 })
+function signOut() {
+  var auth2 = gapi.auth2.getAuthInstance();
```
Comment (at-quangtran) [inline public/js/public/logout.js]: cái này ở đâu ra em?

File: public/js/public/login.js [hunk 3]
```diff
@@ -44,19 +34,60 @@ $(document).ready(function () {
...
       error: function (response) {
         alert(response.responseJSON.error);
         $('.login-form input[type="password"]').val('');
       }
     });
   })
-  name = localStorage.getItem("username");
-  $('#userName').html(name);
 })
+function onSignIn(googleUser) {
```
Comment (at-quangtran) [inline public/js/public/login.js]: googleUser ở đâu ra em?

File: app/Http/Controllers/Api/User/LoginController.php [hunk 4]
```diff
@@ -72,4 +72,36 @@ public function checkLoginToken()
...
+        if (!$user->exists) {
+            $user->username = $request->username;
+            $user->full_name = $request->full_name;
+            $user->birthday = '1996-11-16';
+            $user->gender = 1;
+            $user->phone = 'not found';
+            $user->password = bcrypt('12345678');
+            $user->gplus_id = $request->gplus_id;
+            $user->role_id = 3;
+            $user->phone = 'not found';
```
Comment (at-quangtran) [inline app/Http/Controllers/Api/User/LoginController.php]: bị trùng

File: public/js/public/login.js (hunk 3 above)
Comment (at-vyle) [reply public/js/public/login.js]: https://developers.google.com/identity/sign-in/web/sign-in, em làm theo cái link ni đây ạ

File: public/js/public/logout.js (hunk 2 above)
Comment (at-vyle) [reply public/js/public/logout.js]: https://developers.google.com/identity/sign-in/web/sign-in, em làm theo cái link ni đây ạ

**Review by mvn-quangtran-dn — APPROVED** [review]
LGTM

**Pull request merged**